tracing-appender/src/non_blocking.rs

```diff
@@ -8,14 +8,52 @@ use std::sync::Arc;
 use std::thread::JoinHandle;
 use tracing_subscriber::fmt::MakeWriter;
 
+/// The default maximum number of buffered log lines.
+///
+/// If [NonBlocking][non-blocking] is configured to be lossy, it will drop any additional logs emitted when at
+/// capacity. If it is not lossy, backpressure will be exerted on senders, causing them to wait
+/// until there is buffer capacity remaining before enqueuing new lines.
+///
+/// [non-blocking]: ./struct.NonBlocking.html
+/// Recommended to be a power of 2.
 pub const DEFAULT_BUFFERED_LINES_LIMIT: usize = 128_000;
 
+/// A guard which triggers an associated [`NonBlocking`] writer to flush logs when dropped.
+///
+/// Writing to a [`NonBlocking`] writer will **not** immediately write a log line to the underlying
+/// output. Instead, the log line will be enqueued to be written by the logging worker thread. In
+/// addition, to improve throughput, the non-blocking writer flushes the underlying output
+/// periodically, rather than every time a line is written. This means that if the program
+/// terminates abruptly (such as by panicking, or by calling `std::process::exit`), some log lines
+/// may not be written.
+///
+/// [`NonBlocking`]: ./struct.NonBlocking.html
+/// Since logs recorded near a crash are often necessary for diagnosing the failure, this type
+/// provides a mechanism to ensure that all buffered logs are written to the output, and that the
+/// output is flushed prior to terminating. In order for this to work, this guard should generally
+/// be held in the `main` function (or whatever the outermost scope of the program is). This will
+/// ensure that it is dropped when unwinding, or when `main` returns.
```

Comment:
```suggestion
/// Since spans/events and events recorded near a crash are often necessary for diagnosing the failure, 
/// `WorkerGuard` provides a mechanism to ensure that _all_ buffered logs are flushed to their output.
/// `WorkerGuard` should be assigned in the `main` function or whatever the entrypoint of the program is.
/// This will ensure that the guard will be dropped during an unwinding or when `main` exits
/// successfully.
```